crates/core/tedge_mapper/src/c8y/tests.rs

```diff
@@ -754,6 +729,145 @@ async fn convert_event_with_extra_fields_to_c8y_json() -> Result<()> {
+    );
+
+    let result = converter.convert(&big_measurement_message).await;
+
+    assert!(result
+        .into_iter()
+        .nth(0)
+        .unwrap()
+        .payload_str()
+        .unwrap()
+        .contains(
+            "\"type\":\"ThinEdgeMeasurement\",\"temperature\":{\"temperature\":{\"value\":0.0}}"
+        ));
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_convert_big_measurement_for_child_device() {
+    let mut converter = create_test_c8y_converter();
+
+    let measurement_topic = "tedge/measurements/child1";
+    let big_measurement_payload = create_thin_edge_measurement(10 * 1024); // Measurement payload > size_threshold after converting to c8y json
+
+    let big_measurement_message = Message::new(
+        &Topic::new_unchecked(measurement_topic),
+        big_measurement_payload,
+    );
+
+    let result = converter.convert(&big_measurement_message).await;
+
+    assert!(result
+        .into_iter()
+        .nth(0)
+        .unwrap()
+        .payload_str()
+        .unwrap()
+        .contains("Cumulocity mqtt payload {\"type\":\"ThinEdgeMeasurement\",\"externalSource\":{\"e threshold(16KB) is exceeded"))
+}
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_convert_small_measurement_for_child_device() {
+    let measurement_topic = "tedge/measurements/child1";
+    let big_measurement_payload = create_thin_edge_measurement(20); // Measurement payload size is 20 bytes
+
+    let big_measurement_message = Message::new(
+        &Topic::new_unchecked(measurement_topic),
+        big_measurement_payload,
+    );
+    let mut converter = create_test_c8y_converter();
+    let result = converter.convert(&big_measurement_message).await;
+
+    assert!(result
+        .clone()
+        .into_iter()
+        .nth(0)
+        .unwrap()
+        .payload_str()
+        .unwrap()
+        .contains("101,child1,child1,thin-edge.io-child"));
+
+    assert!(result.clone()
+        .into_iter()
+        .nth(1)
+        .unwrap()
+        .payload_str()
+        .unwrap()
+        .contains(
+            "\"type\":\"ThinEdgeMeasurement\",\"externalSource\":{\"externalId\":\"child1\",\"type\":\"c8y_Serial\"},\"temperature\":{\"temperature\":{\"value\":0.0}}"
+        ));
+}
+
+fn create_packet(size: usize) -> String {
+    let data: String = "Some data!".into();
+    let loops = size / data.len();
+    let mut buffer = String::with_capacity(size);
+    for _ in 0..loops {
+        buffer.push_str("Some data!");
+    }
+    buffer
+}
+
+fn create_thin_edge_measurement(size: usize) -> String {
+    let data = r#""temperature":25"#;
+    let loops = size / data.len();
+    let mut buffer = String::with_capacity(size + 2);
+    buffer.push_str("{");
+    for i in 0..loops {
+        buffer.push_str(&format!("\"temperature\":{}", i));
+        if i != loops - 1 {
+            buffer.push_str(",");
+        }
+    }
+    buffer.push_str("}");
+    buffer
+}
```

Comment:
Another handy way to create this dummy measurement JSON would be to just create a `HashMap<String, serde_json::Value>` with all these different temperature keys and values and just serialize it at the end with `Value::to_string()`. You could avoid all this special character management to properly insert the `,` , `{` and `}`.